manual/opt.tex

```diff
@@ -153,9 +153,108 @@ \subsubsection{quartic}
 \end{lstlisting}
 
 \subsubsection{adaptive}
-%To activate this optimizer, add ``-D BUILD\_LMYENGINE\_INTERFACE=1'' in the CMake command line.
 
-Manual is needed here.
+The default setting of the adaptive optimizer is to construct the linear method Hamiltonian and overlap matrices explicitly and add different shifts to the Hamiltonian matrix 
+as ``stabilizers''. The generalized eigenvalue problem is solved for each shifted Hamiltonian and updates to wave function parameters are obtained. Then a correlated sampling 
+is performed (using the middle shift's updated wave function as the guiding function) for each shift's updated wave function and the initial trial wave function.
+The cost function for these wave functions is compared, and the update corresponding to the best cost function is selected.
+In the next iteration, the median magnitude of the stabilizers is set to that that generated the best update in the current iteration, thus adapting the magnitude of
+the stabilizers automatically.
+
+Thanks to recently developments in excited state variational principles, the adaptive optimizer is also able to optimize individual excited states directly.  
```

Comment:
'recent'.
IMO, it might be better to skip the first clause, and add a reference to the paper at the end of the sentence.